src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -20,42 +20,50 @@
  * @author Fabien Potencier <fabien@symfony.com>
  *
  * @implements \IteratorAggregate<int, \DOMNode>
+ *
+ * @deprecated since Symfony 7.1, use Symfony\Component\DomCrawler\DomCrawler instead
  */
-class Crawler implements \Countable, \IteratorAggregate
+class Crawler implements CrawlerInterface
 {
     protected ?string $uri;
 
     /**
      * The default namespace prefix to be used with XPath and CSS expressions.
      */
-    private string $defaultNamespacePrefix = 'default';
+    protected string $defaultNamespacePrefix = 'default';
 
     /**
      * A map of manually registered namespaces.
      *
      * @var array<string, string>
      */
-    private array $namespaces = [];
+    protected array $namespaces = [];
 
     /**
      * A map of cached namespaces.
      */
-    private \ArrayObject $cachedNamespaces;
+    protected \ArrayObject $cachedNamespaces;
```

Comment:
All those properties that you change from private to protected should be marked as `@internal` to avoid making them part of the API of the Crawler.